tests/RequestTest.php

```diff
@@ -1033,6 +1033,92 @@ public function testGetRefererInvalid()
         $this->assertNull($request->getReferer());
     }
 
+    /**
+     * @expectedException        Brick\Http\Exception\HttpBadRequestException
+     * @expectedExceptionMessage Invalid protocol: invalid_protocol
+     */
+    public function testGetCurrentShouldReturnHttpBadRequestException()
+    {
+        $request = new Request();
+        $_SERVER['SERVER_PROTOCOL'] = 'invalid_protocol';
+        $request->getCurrent();
+    }
+
+    public function testSetFiles()
+    {
+        $request = new Request();
+        $result = $request->setFiles(['./uploaded_file']);
+        $this->assertInstanceOf(Request::class, $result);
+        $this->assertContains('./uploaded_file', $request->getFiles());
+    }
+
+    public function testSetCookiesShouldRemoveCookieHeader()
+    {
+        $request = new Request();
+        $result = $request->setCookies([]);
+        $this->assertCount(0, $result->getCookie());
+    }
+
+    public function testSetRequestUriWithNoQueryString()
+    {
+        $request = new Request();
+        $result = $request->setRequestUri('http://localhost?');
+        $this->assertInstanceOf(Request::class, $result);
+        $this->assertContains('http://localhost?', $result->getRequestUri());
+    }
+
+    public function testGetAcceptShouldReturnEmptyArray()
+    {
+        $request = new Request();
+        $this->assertCount(0, $request->setUrl('http://localhost:8000')->getAccept());
+    }
+
+    public function testIsAjaxShouldReturnFalse()
+    {
+        $request = new Request();
+        $this->assertFalse($request->isAjax());
```

Comment:
Same here, testing just the default value is not really useful, you should also at the very least set the `X-Requested-With: XMLHttpRequest` header and check that it now returns true. The best thing would be to use a data provider that tests a few headers and the expected return value.